### DISCUSSION TITLE
Editing a post result in console error: Store "core/interface" is already registered.

### DISCUSSION THREAD

**Discussion by igorschoester** [woocommerce/woocommerce#45949] [WooCommerce New Product Editor] [answered]
When registering a block server side via `\Automattic\WooCommerce\Admin\Features\ProductBlockEditor\register_block_type_from_metadata`
And client side via `registerProductEditorBlockType` from `@woocommerce/product-editor`.

The block is also active when editing a post in the block editor. Now I could add some logic to prevent the registration there and be done. However, it highlighted a more interesting thing.
When I edit a post in the block editor, the import of `registerProductEditorBlockType` from `@woocommerce/product-editor` alone results in a console error: `Store "core/interface" is already registered.`.

I think this is due to the main entry of `@woocommerce/product-editor` is running code already: https://github.com/woocommerce/woocommerce/blob/d279466eb3274c30baf758e4451aa17a79d3f5c2/packages/js/product-editor/src/index.ts#L44
Though it could very well be that script bundling along the `@wordpress/data` package or something.
I must admit I'm not sure how it gets to the `core/interface` registration as I didn't investigate beyond the import being the problem.

As a side note: we are marking `@woocommerce/product-editor` as an external in our Webpack configuration, i.e. we read from `window.wc.productEditor`. Just preferring the import for the auto-generated dependencies via he `DependencyExtractionWebpackPlugin`.

**Comment by mattsherman** [comment, marked as answer]
@igorschoester Thanks for reporting this. I'll convert this to an issue so that we can figure out a proper fix to this.

It doesn't seem to be related to running code in the main entry of `@woocommerce/product-editor`, because if I comment those out, the error still persists.

I think it has something to do with our importing of one or both of `@wordpress/edit-site` (in core WC) and `@wordpress/edit-post` (in `@woocommerce/product-editor`).

There appears to be a Gutenberg issue that seems related to this: https://github.com/WordPress/gutenberg/issues/54252

If I comment out the enqueuing of `@woocommerce/product-editor` ([here](https://github.com/woocommerce/woocommerce/blob/5fe34385a1b163c1294c1455824ab68c2e47f977/plugins/woocommerce/src/Internal/Admin/WCAdminAssets.php#L281)), then the error does not occur when editing a post. But, of course, then wc-admin pages (Home, Product Editor, etc.) do not load then! 😄 Perhaps one solution is to only enqueue `@woocommerce/product-editor` when on a wc-admin page.

**Reply by igorschoester** [reply]
So until that is fixed in Gutenberg, everyone needs to separate their site/post scripts 😅

**Reply by mattsherman** [reply]
That is my understanding of the issue.
